lisa/tests/base.py

```diff
@@ -868,6 +868,43 @@ class RTATestBundle(FtraceTestBundle):
       the associated task will be ignored in the noise accounting.
     """
 
+    @memoized
+    def get_trace_rtapp_tasks(self, trace):
+        """
+        Get the list of actual rt-app tasks names.
+
+        All rt-app generated tasks have a numerical suffix since fork support
+        has been merged. For forked tasks specifically we can also end up with
+        multiple tasks with the same name specified in the rt-app profile.
+
+        :return: a list of task names matching trace tasks.
+
+        """
+        rtapp_names = []
+        for prefix in self.rtapp_tasks_from_profile:
+            regexp = "^{}(-[0-9]+)?$".format(prefix)
```

Comment:
I think you want: `s/?/*/`

When fork is used you'd get `taskname-$ind-$nr_fork` so the group `(-[0-9]+)` could appear zero, once or twice depends on rt-app version and whether the task was a forked or not.

Reply:
Mmm... that makes me wonder why we added `$ind` for then.

Anyway, if that's the syntax used now... you right, will update it. thanks.

Reply:
I can't remember the details now. But it useful when debugging. Specifically when looking at rt-app debug output which relies on $ind. Log files are created with that name as well. So it helps using the $ind to correlate what you see spilled out by rt-app output, the logs and maybe other bits.

Hmm and I think when forking the same task by 2 other tasks, you wouldn't be able to tell the difference without the $ind.

For example: taskA forks TaskF. TaskB forks TaskF too. Without the $ind you'll get TaskF-$nr_forks which you wouldn't be able to tell they are 2 different threads. So $ind helps to differentiate them as 2 separate tasks.